### ISSUE TITLE
Skipping past the end of an OGG file causes infinite loop.

### ISSUE THREAD

**Issue by a2aaron** [RustAudio/rodio#387]
When I run the following code, "Skipping..." is printed but not "Successfully skipped 31 seconds of audio!". 

```rust
use rodio::{source::Source, Decoder};
use std::fs::File;
use std::io::BufReader;
use std::time::Duration;

fn main() {
    // metronome120.ogg is an ogg file with 15 measures of a 120 BPM click track
    // (hence the file is only 30.0 seconds long).
    let file = BufReader::new(File::open("src/metronome120.ogg").unwrap());
    let source = Decoder::new(file).unwrap();

    println!("Skipping...");
    let _ = source.skip_duration(Duration::from_secs_f32(31.0));
    // The line below will never print, instead the program just locks up on the line above.
    println!("Successfully skipped 31 seconds of audio!");
}
```

Note that this seems to happen only with ogg files. When I try this with a WAV file instead, the program prints the second line as expected. It also only happens when going "out of bounds" of the ogg file--if I only skip 5 seconds, then the program works also.

Here is a zip containing a minimal example: [rodio-bug.zip](https://github.com/RustAudio/rodio/files/6835140/rodio-bug.zip)

I am using version `0.14.0` of Rodio on Windows 10.

**Comment by a2aaron** [comment]
Doing some more testing shows that this doesn't happen with mp3 or flac files, only ogg. Maybe this is a lewton bug?

**Comment by a2aaron** [comment]
I think I figured out the problem, `do_skip_duration` doesn't check if the frame length is zero, which indicates that there's no more data left in the `Source`, causing it to try and skip zero samples forever. I've created a PR to fix this: #388
